db/seeds/00_initial.js

```diff
@@ -0,0 +1,171 @@
+    'Killers',
+    'Killers (live)',
+    'Lightning Strikes Twice',
+    'Look For The Truth',
+    'Lord Of The Flies',
+    'Man On The Edge',
+    'Massacre',
+    'Moonchild',
+    'Mother Russia',
+    'Murders In The Rue Morgue',
+    'My Generation',
+    'New Frontier',
+    'No More Lies',
+    'No Prayer For The Dying',
+    'Only The Good Die Young',
+    'Out Of The Silent Planet',
+    'Paschendale',
+    'Phantom Of The Opera',
+    'Powerslave',
+    'Prodigal Son',
+    'Prowler',
+    'Public Enemy Number One',
+    'Purgatory',
+    'Quest For Fire',
+    'Rainmaker',
+    'Remember Tomorrow',
+    'Remember Tomorrow (live)',
+    'Revelations',
+    'Rime Of The Ancient Mariner',
+    'Run Silent Run Deep',
+    'Run To The Hills',
+    'Running Free',
+    'Running Free (live)',
+    'Sanctuary',
+    'Sea Of Madness',
+    'Seventh Son Of A Seventh Son',
+    'Sign Of The Cross',
+    'Still Life',
+    'Strange World',
+    'Stranger In A Strange Land',
+    'Sun And Steel',
+    'Tailgunner',
+    'The Aftermath',
+    'The Angel And The Gambler',
+    'The Apparition',
+    'The Assassin',
+    'The Clairvoyant',
+    'The Duellists',
+    'The Edge Of Darkness',
+    'The Educated Fool',
+    'The Evil That Men Do',
+    'The Fallen Angel',
+    'The Fugitive',
+    'The Loneliness Of The Long Distance Runner',
+    'The Mercenary',
+    'The Nomad',
+    'The Number Of The Beast',
+    'The Prisoner',
+    'The Prophecy',
+    'The Thin Line Between Love And Hate',
+    'The Trooper',
+    'The Unbeliever',
+    'The Wicker Man',
+    'To Tame A Land',
+    'Total Eclipse',
+    'Twilight Zone',
+    'Virus',
+    'Wasted Years',
+    'Wasting Love',
+    'Weekend Warrior',
+    'When Two Worlds Collide',
+    'Where Eagles Dare',
+    'Wildest Dreams',
+    'Women In Uniform',
+    'Wrathchild'
+  ];
+
+  var gotPhotos = await axios.get(`https://api.unsplash.com/search/photos/?query=music&client_id=${key.api_key}`)
+  .then((res) => {
+    return res.data.results
+  }).catch((err) => {
+    console.log('Error in retrieving photos yo');
+    console.log(err);
+  });
+
+  var photos = gotPhotos.map(photo => {
+    return photo.urls.raw
+  });
+
+  // Average Song length on SoundCloud is 3-5 minutes
+  var min = Math.ceil(180); // 3 minutes converted to seconds
+  var max = Math.floor(300); // 5 minutes
+
+  var scaledData = 10000000;
+
+  for (var i = 1; i <= scaledData; i++) {
+    var id = i;
+    var dataToSave = { song_id: id };
+    dataToSave.song_length = Math.floor(Math.random() * (max - min) + min);
+    dataToSave.song_name = names[Math.floor(Math.random() * ((names.length - 1)))];
+    dataToSave.song_url = 'https://rpt23-fec-soundcloud.s3-us-west-2.amazonaws.com/Djenty+Metal+Town%2C+USA.mp3'
+    dataToSave.song_image = photos[Math.floor(Math.random() * ((photos.length - 1)))];
+    // dataToSave.bandID = Math.floor(Math.random() * (30 - 1) + 1);
+    var saved = await knex('songs').insert(dataToSave);
```

Comment:
Looks like you are loading data here into your DB while looping through 10M times. Have you attempted to generate all Data separately, and load at once on your target DB per instructions on learn per section "DBMS Selection and Data Generation". The language on learn does suggest you need to attempt, and if you have already, don't consider this as gating to the PR, and respond accordingly. If you haven't I would suggesting giving it a go, if time permits.

Reply:
For sure. Would that be like writing it all to a csv file and then saving it to the db like Toly did?

Reply:
Essentially yes, CSV will help flatten it and reduce the space needed to store the data, as opposed to storing JS/JSON. Benefit of that is you would be able to use the same data to load in any DB, so your generation script is separate from your loading script/code.

However, considering language on learn, recommendation is to attempt, and I dont see that as a hard requirement. Give it a go, and if you deem it fit to continue with what you have, please respond as such, I am ok if you are. Caveat is you will need to take similar approach for you secondary DB.